feat(apple): Explain app hang algorithm (#6717)
getsentry/sentry-docs: src/platforms/apple/common/configuration/app-hangs.mdx

> There are many reasons an app can become unresponsive, from long running code to an infinite loop bug, and this can be frustrating to the end user.
> With app hang tracking, you can detect and fix this problem.

Reviewer: ```suggestion
Trying to use an unresponsive app is extremely frustrating for users. There are many reasons why an app may become unresponsive, such as long- running code, infinite loop bugs, and so on. With app hang tracking you can detect and fix them.
```

> The app hang detection integration has a default timeout of two (2) seconds. With 7.24.0, we've improved the algorithm to detect app hangs sooner, decreasing it from `appHangTimeoutInterval * 2` to `appHangTimeoutInterval * 1.2`. The SDK creates an error event if the app is unresponsive for two seconds or more.
> For detection, the SDK launches a dedicated watchdog thread, periodically enqueues work items on the main thread, and checks if the main thread executes them within the timeout.
> Recording the stacktrace precisely when the app hang occurs works reliably if the app is completely stuck. If the main thread is extremely busy with different code spots, our app hang detection might fire a bit too late, and the SDK might record a stacktrace that isn't 100% related to the code causing the app to hang.

Reviewer: ```suggestion

The app hang detection integration has a default timeout of two (2) seconds, but with version 7.24.0, we’ve improved the algorithm to detect app hangs sooner, decreasing from `appHangTimeoutInterval * 2` to `appHangTimeoutInterval * 1.2`. 

To detect app hangs, the SDK launches a dedicated watchdog thread, periodically enqueues work items on the main thread, and checks if the main thread is executing them within the timeout. If the app is unresponsive for two seconds or more, it creates an error event. 

Recording the stack trace precisely when the app hang occurs works reliably if the app is completely stuck, but if the main thread is extremely busy with different code spots, the app hang detection might fire a bit too late. If this happens, the SDK may record a stack trace that isn’t 100% related to the code, causing the app to hang. Each event has a stack trace of all running threads so you can easily detect where the problem occurred.

```

> The event has the stack trace of all running threads so you can easily detect where the problem occurred. The SDK reports an app hang immediately but doesn't report the exact duration because the [watchdog](https://developer.apple.com/documentation/xcode/addressing-watchdog-terminations) could kill the app any time when blocking the main thread.

Reviewer: ```suggestion
The SDK reports an app hang immediately, but doesn’t report the exact duration because the [watchdog](https://developer.apple.com/documentation/xcode/addressing-watchdog-terminations) could kill the app anytime if it's blocking the main thread.
```
